cpp/src/arrow/compute/kernels/vector_run_end_encode_test.cc

```diff
@@ -0,0 +1,196 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
Are the additional kernels (encode/decode, filter) required for this PR? Otherwise, it would be better to file them as separate PRs.

Reply:
No, not strictly required. When I probe for the next round of reviews I will push with a smaller set of commits.